cpp/src/join/distinct_hash_join.cu

```diff
@@ -214,14 +297,23 @@ std::unique_ptr<rmm::device_uvector<size_type>> distinct_hash_join<HasNested>::l
   } else {
     auto const probe_row_hasher =
       cudf::experimental::row::hash::row_hasher{this->_preprocessed_probe};
-    auto const d_probe_hasher = probe_row_hasher.device_hasher(nullate::DYNAMIC{this->_has_nulls});
-    auto const iter           = cudf::detail::make_counting_transform_iterator(
-      0, build_keys_fn<decltype(d_probe_hasher), lhs_index_type>{d_probe_hasher});
-
-    auto const output_begin =
-      thrust::make_transform_output_iterator(build_indices->begin(), output_fn{});
-    // TODO conditional find for nulls once `cuco::static_set::find_if` is added
-    this->_hash_table.find_async(iter, iter + probe_table_num_rows, output_begin, stream.value());
+    auto const d_probe_hasher =
+      probe_row_hasher.device_hasher(probe_column_types, nullate::DYNAMIC{this->_has_nulls});
+
+    std::visit(
+      [&](auto&& hasher, auto&& hash_table) {
+        auto const iter = cudf::detail::make_counting_transform_iterator(
+          0,
+          build_keys_fn<typename std::remove_reference_t<decltype(hasher)>, lhs_index_type>{
+            hasher});
+
+        auto const output_begin =
+          thrust::make_transform_output_iterator(build_indices->begin(), output_fn{});
+        // TODO conditional find for nulls once `cuco::static_set::find_if` is added
```

Comment:
This feature now exists in cuCollections, I think. Let's refactor if we can.

Reply:
I think maybe we could address this in a separate MR since the change wouldn't reflect this MR description. What do you think?